lxd-agent/main_agent.go

```diff
@@ -250,7 +253,12 @@ func (c *cmdAgent) writeStatus(status string) error {
 		if err != nil {
 			return err
 		}
-		vSerial.Write([]byte(fmt.Sprintf("%s\n", status)))
+
+		_, err = vSerial.Write([]byte(fmt.Sprintf("%s\n", status)))
+		if err != nil {
+			return err
+		}
+
 		vSerial.Close()
```

Comment:
This probably needs to be excluded from the lint ignore list, as that Close may need to write something and could fail.